### ISSUE TITLE
Fix Flacky Tests

### ISSUE THREAD

**Issue by ns-vasilev** [space-code/flare#77]
This issue is meant to be a good first issue to get into coding, and into open source contribution over all, as it will require to touch few existing files, is a small change and is documented extensively. If you are new to coding, I encourage you to try this issue.

Please read our [contribution guidelines](https://github.com/space-code/flare/blob/dev/CONTRIBUTING.md) before picking this up, it will also lead you through the process of setting up the repository.

**Feature description**
Fix the integration test `test_thatFlarePurchasesAProductWithCompletion_whenUnkownErrorOccurred` so it doesn't flake and the job doesn't fail in GitHub Actions.

**Comment by M7md-Ebrahim** [comment]
Excited to fix this integration test! Can you assign it to me?

**Comment by nik3212** [comment]
@M7md-Ebrahim, sure

**Comment by nik3212** [comment]
@M7md-Ebrahim how's it going? Do you want to continue on this?

**Comment by M7md-Ebrahim** [comment]
Yes, I’m a bit busy, and I’m not very good at testing. If you don’t mind, I’ll give it another try.

**Comment by mcrkgus** [comment]
I’d like to tackle with this issue. is there a dead line? or is this issue already resolved?

**Comment by ns-vasilev** [comment]
@mcrkgus, hi! This issue has not been resolved yet. Of course, you can tackle it if you wish.
